src/init.cc

```diff
@@ -1118,140 +1141,163 @@ static ncclResult_t initTransportsRank(struct ncclComm* comm, struct ncclComm* p
   }
   comm->topParentLocalRanks = topParentLocalRanks;
 
+  NCCLCHECKGOTO(ncclTransportCheckP2pType(comm, &comm->intraNodeP2pSupport, &comm->directMode), ret, fail);
   // Launch proxy service thread, after this, the proxy calls can be used.
-  NCCLCHECKGOTO(ncclProxyCreate(comm), ret, fail);
-
-  // Connect with prev/next for each ring
-  for (int c=0; c<comm->nChannels; c++) {
-    struct ncclChannel* channel = comm->channels+c;
-    NCCLCHECKGOTO(setupChannel(comm, c, rank, nranks, rings+c*nranks), ret, fail);
-    if (comm->nRanks == 1) continue;
-    NCCLCHECKGOTO(ncclTransportP2pConnect(comm, c, 1, &channel->ring.prev, 1, &channel->ring.next, 0), ret, fail);
-  }
-  NCCLCHECKGOTO(ncclTransportP2pSetup(comm, &ringGraph, 0, &highestTransportType, &needsProxy), ret, fail);
-  mscclHighestTransportType = std::max(mscclHighestTransportType, highestTransportType);
-  mscclNeedsProxy |= needsProxy;
```

Comment:
`mscclHighestTransportType` and `mscclNeedsProxy` needs to be probed properly.

`mscclHighestTransportType` was used to judge whether a connection is via NIC which decides whether to use extra memory fences in MSCCL kernels. Missing it will cause data error and hang in cross-node MSCCL collectives.

`mscclNeedsProxy` was used to load proxy ops properly for MSCCL kernels. Missing it will cause hang because proxy threads will not move data for cross-node MSCCL collectives.

Reply:
We can add `highestTransportType` and `needsProxy` args in `ncclTransportP2pSetup` in `msccl-executor-nccl/src/transport/generic.cc`.

Reply:
Resolved in latest commit.